src/python/pants/backend/docker/util_rules/docker_build_context.py

```diff
@@ -271,10 +272,20 @@ async def create_docker_build_context(
         for field_set in embedded_pkgs_per_target.field_sets
         # Exclude docker images, unless build_upstream_images is true.
         if request.build_upstream_images
-        or not isinstance(getattr(field_set, "sources", None), DockerImageSourceField)
+        or not isinstance(getattr(field_set, "source", None), DockerImageSourceField)
     )
 
-    packages_str = ", ".join(a.relpath for p in embedded_pkgs for a in p.artifacts if a.relpath)
+    images_str = (
+        ", ".join(
+            a.tags[0] for p in embedded_pkgs for a in p.artifacts if isinstance(a, BuiltDockerImage)
+        )
+        or "<None>"
+    )
+    logger.debug(f"Built upstream Docker images: {images_str}")
```

Comment:
How about instead logging "Did not build any upstream Docker images" if none are necessary? Looks like you can check `request.build_upstream_images`? Only create `images_str` if relevant.